pkg/sql/alter_table_set_schema.go

```diff
@@ -0,0 +1,179 @@
+// Copyright 2020 The Cockroach Authors.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package sql
+
+import (
+	"context"
+
+	"github.com/cockroachdb/cockroach/pkg/kv"
+	"github.com/cockroachdb/cockroach/pkg/sql/catalog/catalogkv"
+	"github.com/cockroachdb/cockroach/pkg/sql/pgwire/pgcode"
+	"github.com/cockroachdb/cockroach/pkg/sql/pgwire/pgerror"
+	"github.com/cockroachdb/cockroach/pkg/sql/privilege"
+	"github.com/cockroachdb/cockroach/pkg/sql/sem/tree"
+	"github.com/cockroachdb/cockroach/pkg/sql/sqlbase"
+	"github.com/cockroachdb/cockroach/pkg/util/log"
+)
+
+type alterTableSetSchemaNode struct {
+	newSchema string
+	tableDesc *sqlbase.MutableTableDescriptor
+	n         *tree.AlterTableSetSchema
+}
+
+// AlterTableSetSchema sets the schema for a table, view or sequence.
+// Privileges: DROP on source table/view/sequence, CREATE on destination schema.
+func (p *planner) AlterTableSetSchema(
+	ctx context.Context, n *tree.AlterTableSetSchema,
+) (planNode, error) {
+	tn := n.Name.ToTableName()
+	requiredTableKind := tree.ResolveAnyTableKind
```

Comment:
Why do you need the kind? Can't you just use tree.ResolveRequireAny (or whatever the default value is)